W-13687911-amazon-sqs-regionalized-kt (#2345)
mulesoft/docs-connectors: amazon-sqs/5.11/modules/ROOT/pages/amazon-sqs-connector-troubleshooting.adoc

include::connectors::partial$verbose-logs-org.adoc[]

== Troubleshoot AWS STS Regionalized Endpoints

> Since the upgrade of AWS SDK from v1 to v2, the endpoint that the STS client of the AssumeRoleConnectionProvider uses to get session tokens now is Regionalized instead of Global. Refer to the https://docs.aws.amazon.com/sdkref/latest/guide/feature-sts-regionalized-endpoints.html[AWS STS Regionalized endpoints] documentation for details.

Reviewer: In AWS SDK v2, the endpoint that the STS client of the `AssumeRoleConnectionProvider` operation uses to get session tokens is **Regionalized** instead of **Global**. See the https://docs.aws.amazon.com/sdkref/latest/guide/feature-sts-regionalized-endpoints.html[AWS STS Regionalized endpoints] documentation for details.

> Since the upgrade of AWS SDK from v1 to v2, the endpoint that the STS client of the AssumeRoleConnectionProvider uses to get session tokens now is Regionalized instead of Global. Refer to the https://docs.aws.amazon.com/sdkref/latest/guide/feature-sts-regionalized-endpoints.html[AWS STS Regionalized endpoints] documentation for details.

Reviewer: ```suggestion
In AWS SDK v2, the endpoint that the STS client of the `AssumeRoleConnectionProvider` operation uses to get session tokens is now *Regionalized* instead of *Global*. See the https://docs.aws.amazon.com/sdkref/latest/guide/feature-sts-regionalized-endpoints.html[AWS STS Regionalized endpoints] documentation for details.
```

> Additionally, as explained in the https://docs.aws.amazon.com/general/latest/gr/sts.html[AWS Security Token Service endpoints and quotas] documentation, regionalized endpoints shouldn't represent any change in the connector's functionality.

Reviewer: As explained in the https://docs.aws.amazon.com/general/latest/gr/sts.html[AWS Security Token Service endpoints and quotas] documentation, regionalized endpoints shouldn’t cause changes in the connector's functionality.

> Additionally, as explained in the https://docs.aws.amazon.com/general/latest/gr/sts.html[AWS Security Token Service endpoints and quotas] documentation, regionalized endpoints shouldn't represent any change in the connector's functionality.

Reviewer: ```suggestion
As explained in the https://docs.aws.amazon.com/general/latest/gr/sts.html[AWS Security Token Service endpoints and quotas] documentation, regionalized endpoints shouldn’t cause changes in the connector's functionality.
```

> AWS recommends using Regional STS endpoints to reduce latency, build in redundancy, and increase session token validation.

Reviewer: AWS recommends using Regional STS endpoints to reduce latency, build in redundancy, and increase session token validation.

> AWS recommends using Regional STS endpoints to reduce latency, build in redundancy, and increase session token validation.

Reviewer: ```suggestion
AWS recommends using Regional STS endpoints to reduce latency, build in redundancy, and increase session token validation.
```

> If you encounter any issues about the endpoints change in the regions, you can force Amazon SQS Connector to use the global endpoint by specifying the URL endpoint https://sts.amazonaws.com in the Custom STS Endpoint field located in the Advanced tab of the Global Elements properties configuration window of the connector.

Reviewer: If you encounter any issues as a result of the endpoints change in the regions, you can force Amazon SQS Connector to use the global endpoint by specifying the URL endpoint \https://sts.amazonaws.com in the **Custom STS Endpoint** field located in the **Advanced** tab of the **Global Elements** properties configuration window of the connector.

Reviewer: Is that how you escape a URL? \ ? If not, please escape the URL so it's not a live link.

> If you encounter any issues about the endpoints change in the regions, you can force Amazon SQS Connector to use the global endpoint by specifying the URL endpoint https://sts.amazonaws.com in the Custom STS Endpoint field located in the Advanced tab of the Global Elements properties configuration window of the connector.

Reviewer: ```suggestion
If you encounter any issues as a result of the change in endpoints to regionalized, you can force Amazon SQS Connector to use the global endpoint by specifying the URL endpoint \https://sts.amazonaws.com in the **Custom STS Endpoint** field located in the **Advanced** tab of the **Global Elements** properties configuration window of the connector.
```

Reviewer: Is that how you escape a URL? \ ? If not, please escape the URL so it's not a live link.
